MOAction/MOAction.cs

```diff
@@ -239,109 +225,110 @@ private unsafe (Lumina.Excel.GeneratedSheets.Action action, GameObject target) G
+                if (action.TargetArea) return (true, target);
+                else if (err != 0 && err != 565) return (false, target);
             }
-            if (target.ObjectKind == ObjectKind.Player) return action.CanTargetFriendly ||
-                    action.CanTargetParty ||
-                    action.CanTargetSelf ||
-                    action.TargetArea ||
-                    UnorthodoxFriendly.Contains((uint)action.RowId);
-            if (target.ObjectKind == ObjectKind.BattleNpc)
+
+            //Skip actions you cannot use when synced down, excluding role actions that can be used even when syncing down.
+            pluginLog.Verbose("is {actionname} a role action?: {answer}", action.Name, action.IsRoleAction);
+            if (!action.IsRoleAction)
             {
-                BattleNpc b = (BattleNpc)target;
-                if (!(b.BattleNpcKind == BattleNpcSubKind.Enemy || b.BattleNpcKind == BattleNpcSubKind.BattleNpcPart)){
-                    return action.CanTargetFriendly ||
-                        action.CanTargetParty ||
-                        action.CanTargetSelf ||
-                        action.TargetArea ||
-                        UnorthodoxFriendly.Contains((uint)action.RowId);
-                }
+                pluginLog.Verbose("is {actionName} usable at level: {level} available for player {playername} with {playerlevel}?", action.Name, action.ClassJobLevel, player.Name, player.Level);
+                if (action.ClassJobLevel > clientState.LocalPlayer.Level) return (false, target);
             }
-            return action.CanTargetHostile ||
-                action.TargetArea ||
-                UnorthodoxHostile.Contains((uint)action.RowId);
+
+            //area of effect spells do not require a target to work.
+            pluginLog.Verbose("is {actionname} a area spell/ability? {answer}", action.Name, action.TargetArea);
+            if (action.TargetArea) return (true, target);
+
+
+
+            //sanity check on spells that can not be used on friendly, hostile or party
+            bool selfOnlyTargetAction = !action.CanTargetFriendly && !action.CanTargetHostile && !action.CanTargetParty;
+            pluginLog.Verbose("Can {actionname} target: friendly - {friendly}, hostile  - {hostile}, party  - {party}, dead - {dead}, self - {self}", action.Name, action.CanTargetFriendly, action.CanTargetHostile, action.CanTargetParty, action.CanTargetDead, action.CanTargetSelf);
+            if (selfOnlyTargetAction)
+            {
+                pluginLog.Verbose("Can only use this action on player, setting player as target");
+                target = clientState.LocalPlayer;
+            }
+
+            //handoff to game native code, returns true if the actionManager declares that the action can be used on the target specified
+            bool gameCanUseActionResponse = ActionManager.CanUseActionOnTarget(action.RowId, (FFXIVClientStructs.FFXIV.Client.Game.Object.GameObject*)target.Address);
```

Comment:
I'm guessing this is the secret sauce that eliminates the need for the `unorthodox*` data? Those were always a conceptual pain.

Reply:
that eliminated so so so so so much,
handing the decision over to the game in a way we want (native code), we use the game's decisionmaking to decide. killing all the unorthodox AND fixing the quest instance bug